liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,500 @@
+        } is ${data.tokens[currentToken].price.toString()
+        } ETH with a collateral factor of ${data.tokens[currentToken].collateralMult}`,
+      );
+    }));
+    // #endregion
+
+    // #region Calculate health on all accounts
+    ts('Health recalculation');
+    Object.keys(data.accounts).forEach((account) => {
+      let borrowBalance = BigNumber(0);
+      let supplyBalance = BigNumber(0);
+      Object.keys(data.tokens).forEach((token) => {
+        if (data.supply[token][account]) {
+          // Supply balances are stored in cTokens and need to be multiplied by the
+          //   exchange rate and reduced by the collateral factor
+          supplyBalance = supplyBalance.plus(data.supply[token][account]
+            .multipliedBy(data.tokens[token].exchangeRateMult)
+            .multipliedBy(data.tokens[token].price)
+            .multipliedBy(data.tokens[token].collateralMult));
+        }
+        if (data.borrow[token][account]) {
+          // Only need to multiply by the price.
+          borrowBalance = borrowBalance.plus(data.borrow[token][account]
+            .multipliedBy(data.tokens[token].price));
+        }
+      });
+
+      // Remove non-borrowers
+      if (borrowBalance.eq(0)) {
+        /* eslint-disable no-param-reassign */
+        delete data.accounts[account];
+      } else {
+        data.accounts[account].supplyBalance = supplyBalance;
+        data.accounts[account].borrowBalance = borrowBalance;
+        data.accounts[account].health = supplyBalance.dividedBy(borrowBalance);
+        /* eslint-enable no-param-reassign */
+      }
+    });
+    // #endregion
+
+    // #region Check for low health accounts currently health <1.03
+    const lowHealthAccounts = [];
+    Object.keys(data.accounts).forEach((currentAccount) => {
+      if (data.accounts[currentAccount].health.isLessThan(1.03)) {
+        lowHealthAccounts.push(currentAccount);
+      }
+    });
+    ts(`${lowHealthAccounts.length} low health accounts detected`);
+    await Promise.all(lowHealthAccounts.map(async (currentAccount) => {
+      const liquidity = await data.comptrollerContract.getAccountLiquidity(currentAccount);
+      const positiveUSD = BigNumber(ethers.utils.formatEther(liquidity[1]).toString());
+      const shortfallUSD = BigNumber(ethers.utils.formatEther(liquidity[2]).toString());
+
+      if (positiveUSD.gt(0)) {
+        // Uncomment this if you want to see low liquidity as well.
+        // ts(
+        //   `Low liquidity on ${currentAccount
+        //   } with health of ${data.accounts[currentAccount].health.dp(3)
+        //   } and cash reserve of +$${positiveUSD.dp(2)} USD`,
+        // );
+      }
+      if (shortfallUSD.gt(0)) {
+        ts(
+          `Negative liquidity on ${currentAccount} with health of ${data.accounts[currentAccount].health.dp(3)
+          } and shortfall of +$${shortfallUSD.dp(2)} USD`,
+        );
+      }
+      // Lower health means that less money that can be retrieved.
```

Comment:
Is this comment accurate?

Isn't health the ratio between amount supplied and amount borrowed?  So a low ratio doesn't indicate anything about the amount that can be retrieved.

Reply:
Corrected in commit 79512957480ad93f994c11aa269380233ad83787